Snowflake.Data/Core/FileTransfer/StorageClient/SFSnowflakeAzureClient.cs

```diff
@@ -330,13 +335,14 @@ public async Task DownloadFileAsync(SFFileMetadata fileMetadata, string fullDstP
             fileMetadata.resultStatus = ResultStatus.DOWNLOADED.ToString();
         }
 
-        private SFFileMetadata HandleFileHeaderErr(RequestFailedException ex, SFFileMetadata fileMetadata)
+        private SFFileMetadata HandleFileHeaderErr(Exception ex, SFFileMetadata fileMetadata)
         {
-            if (ex.Status == (int)HttpStatusCode.BadRequest)
+            RequestFailedException err = (RequestFailedException)ex;
```

Comment:
singe it is cast here to RequestFailedException are we sure it will always be this type or maybe we can revert this type change at all?

Reply:
I'll revert the type change